File: mRemoteV1/CS/Container/Container.List.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports


namespace mRemoteNG.Container
{
	public class List : CollectionBase
		{
			
#region Public Properties
public Container.Info this[object Index]
			{
				get
				{
					if (Index is Container.Info)
					{
						return Index;
					}
					else
					{
						return ((Container.Info) (List[System.Convert.ToInt32(Index)]));
					}
				}
			}
			
public new int Count
			{
				get
				{
					return List.Count;
				}
			}
#endregion
			
#region Public Methods
			public Container.Info Add(Container.Info cInfo)
			{
				this.List.Add(cInfo);
				return cInfo;
			}
			
			public void AddRange(Container.Info[] cInfo)
			{
				foreach (Container.Info cI in cInfo)
				{
					List.Add(cI);
				}
			}
			
			public Container.Info FindByConstantID(string id)
			{
				foreach (Container.Info contI in List)
				{
					if (contI.ConnectionInfo.ConstantID == id)
					{
						return contI;
					}
				}
				
				return null;
			}
			
			public Container.List Copy()
			{
				try
				{
					return this.MemberwiseClone();
				}
				catch (Exception)
				{
				}
				
				return null;
			}
			
			public new void Clear()
			{
				this.List.Clear();
			}
#endregion
		}
}

File: mRemoteV1/Container/Container.List.vb
Namespace Container
    Public Class List
        Inherits CollectionBase

#Region "Public Properties"

        Default Public ReadOnly Property Items(Index As Object) As Info
            Get
                If TypeOf Index Is Info Then
                    Return Index
                Else
                    Return CType(List.Item(Index), Info)
                End If
            End Get
        End Property

        Public Shadows ReadOnly Property Count As Integer
            Get
                Return List.Count
            End Get
        End Property

#End Region

#Region "Public Methods"

        Public Function Add(cInfo As Info) As Info
            Me.List.Add(cInfo)
            Return cInfo
        End Function

        Public Sub AddRange(cInfo() As Info)
            For Each cI As Info In cInfo
                List.Add(cI)
            Next
        End Sub

        Public Function FindByConstantID(id As String) As Info
            For Each contI As Info In List
                If contI.ConnectionInfo.ConstantID = id Then
                    Return contI
                End If
            Next

            Return Nothing
        End Function

        Public Function Copy() As List
            Try
                Return Me.MemberwiseClone
            Catch ex As Exception
            End Try

            Return Nothing
        End Function

        Public Shadows Sub Clear()
            Me.List.Clear()
        End Sub

#End Region
    End Class
End Namespace